### ISSUE TITLE
Don't require docstrings on inner functions

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#1956]
### Steps to reproduce

Run pylint on the following code:
```python
def outer():
    """Documented."""
    def inner():
        # Not documented.
        return 42
    return inner()
```

### Current behavior

The following issue is reported:
```
C:  3, 4: Missing function docstring (missing-docstring)
```

### Expected behavior

Inner functions are not part of any public interface, so if they need documentation, that can be done using comments instead of docstrings. Therefore I think PyLint does not need to warn about missing docstrings on inner functions.

### pylint --version output

```
No config file found, using default configuration
pylint 1.8.2, 
astroid 1.6.1
Python 2.7.13 (default, Jan 03 2017, 17:41:54) [GCC]
```

**Comment by PCManticore** [comment]
Sounds good to me.

**Comment by glmdgrielson** [comment]
You mean this only _just now_ came up? I'll find a solution if need be because that seems **really** arbitrary!
There's no way to get to them from the outside so why are they necessary?

**Comment by PCManticore** [comment]
What's with the passive aggressiveness?

**Comment by glmdgrielson** [comment]
Sorry, I just overreacted. It seemed unbelievable to imagine a problem like only coming up `9 days ago`. I have provided a fix though.
